solver/internal/pipe/pipe_test.go

```diff
@@ -88,5 +89,5 @@ func TestPipeCancel(t *testing.T) {
 	require.Equal(t, st.Completed, true)
 	require.Equal(t, st.Canceled, true)
 	require.Error(t, st.Err)
-	require.Equal(t, st.Err, context.Canceled)
+	require.True(t, errors.Is(st.Err, context.Canceled))
```

Comment:
nit: there is an [ErrorIs](https://pkg.go.dev/github.com/stretchr/testify/require#ErrorIs) now:
```
require.ErrorIs(t, err, context.Canceled)
```